Bottom: 561ff6bba7e5232274b5acbf7b6a16d34b400711
Top:    1d7c9857c0b8be7dc31593da4d2647068d4cc1ce
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-11-21 21:09:38 +0000

Refresh of push-special-case-thindraw-vs-

---

diff --git a/src/hid/common/draw_helpers.c b/src/hid/common/draw_helpers.c
index ddd7294..96db6cc 100644
--- a/src/hid/common/draw_helpers.c
+++ b/src/hid/common/draw_helpers.c
@@ -4,6 +4,7 @@
 #include "data.h" /* For nasty global "PCB" variable */
 #include "rotate.h" /* For RotateLineLowLevel() */
 #include "polygon.h"
+#include "draw_helpers.h"
 
 
 static void
@@ -241,12 +242,12 @@ should_compute_no_holes (PolygonType *poly, const BoxType *clip_box)
 #undef BOUNDS_INSIDE_CLIP_THRESHOLD
 
 void
-common_gui_draw_pcb_polygon (hidGC gc, PolygonType *poly, const BoxType *clip_box)
+common_gui_draw_pcb_polygon (hidGC gc, PolygonType *polygon, const BoxType *clip_box)
 {
   if (TEST_FLAG (THINDRAWFLAG, PCB) || TEST_FLAG (THINDRAWPOLYFLAG, PCB))
-    common_thindraw_pcb_polygon (gc, polygon, drawn_area);
+    common_thindraw_pcb_polygon (gc, polygon, clip_box);
   else
-    common_fill_pcb_polygon (gc, polygon, drawn_area);
+    common_fill_pcb_polygon (gc, polygon, clip_box);
 
   /* If checking planes, thin-draw any pieces which have been clipped away */
   if (TEST_FLAG (CHECKPLANESFLAG, PCB) && !TEST_FLAG (FULLPOLYFLAG, polygon))
@@ -256,7 +257,7 @@ common_gui_draw_pcb_polygon (hidGC gc, PolygonType *poly, const BoxType *clip_bo
       for (poly.Clipped = polygon->Clipped->f;
            poly.Clipped != polygon->Clipped;
            poly.Clipped = poly.Clipped->f)
-        common_thindraw_pcb_polygon (gc, &poly, drawn_area);
+        common_thindraw_pcb_polygon (gc, &poly, clip_box);
     }
 }
 
diff --git a/src/hid/common/draw_helpers.h b/src/hid/common/draw_helpers.h
index 673b248..1b59945 100644
--- a/src/hid/common/draw_helpers.h
+++ b/src/hid/common/draw_helpers.h
@@ -3,6 +3,6 @@ void common_fill_pcb_polygon (hidGC gc, PolygonType *poly, const BoxType *clip_b
 void common_thindraw_pcb_polygon (hidGC gc, PolygonType *poly, const BoxType *clip_box);
 void common_fill_pcb_pad (hidGC gc, PadType *pad, bool clear, bool mask);
 void common_thindraw_pcb_pad (hidGC gc, PadType *pad, bool clear, bool mask);
-void common_fill_pcb_pv (hidGC gc, PinType *pv, bool drawHole, bool mask);
+void common_fill_pcb_pv (hidGC fg_gc, hidGC bg_gc,  PinType *pv, bool drawHole, bool mask);
 void common_thindraw_pcb_pv (hidGC fg_gc, hidGC bg_gc, PinType *pv, bool drawHole, bool mask);
 void common_draw_helpers_init (HID_DRAW *graphics);
diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index 234c061..8e39be4 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -2181,8 +2181,6 @@ hid_gtk_init ()
   ghid_hid.notify_save_pcb          = ghid_notify_save_pcb;
   ghid_hid.notify_filename_changed  = ghid_notify_filename_changed;
 
-  ghid_hid.draw_pcb_polygon         = common_gui_draw_pcb_polygon;
-
   ghid_hid.graphics                 = &ghid_graphics;
 
   ghid_graphics.make_gc             = ghid_make_gc;
@@ -2199,6 +2197,8 @@ hid_gtk_init ()
   ghid_graphics.fill_polygon        = ghid_fill_polygon;
   ghid_graphics.fill_rect           = ghid_fill_rect;
 
+  ghid_graphics.draw_pcb_polygon    = common_gui_draw_pcb_polygon;
+
   hid_register_hid (&ghid_hid);
 #include "gtk_lists.h"
 }
